Source/WebCore/dom/Node.cpp

```diff
@@ -2521,27 +2523,31 @@ void Node::defaultEventHandler(Event& event)
         }
 #endif
     } else if (eventType == eventNames.textInputEvent) {
-        if (is<TextEvent>(event)) {
+        if (RefPtr textEvent = dynamicDowncast<TextEvent>(event)) {
             if (RefPtr frame = document().frame())
-                frame->eventHandler().defaultTextInputEventHandler(downcast<TextEvent>(event));
+                frame->eventHandler().defaultTextInputEventHandler(*textEvent);
         }
 #if ENABLE(PAN_SCROLLING)
-    } else if (eventType == eventNames.mousedownEvent && is<MouseEvent>(event)) {
-        if (downcast<MouseEvent>(event).button() == MouseButton::Middle) {
+    } else if (auto* mouseEvent = dynamicDowncast<MouseEvent>(event); mouseEvent && eventType == eventNames.mousedownEvent) {
+        if (mouseEvent->button() == MouseButton::Middle) {
             if (enclosingLinkEventParentOrSelf())
                 return;
 
-            RenderObject* renderer = this->renderer();
-            while (renderer && (!is<RenderBox>(*renderer) || !downcast<RenderBox>(*renderer).canBeScrolledAndHasScrollableArea()))
-                renderer = renderer->parent();
-
-            if (renderer) {
+            RenderBox* renderBox = nullptr;
+            for (RenderObject* renderer = this->renderer(); renderer; renderer = renderer->parent()) {
```

Comment:
It seems like there should be an iterator for this (like parentsOfType) that handles the traversal of parents. But maybe that only exists for Nodes.